blogs/2021/01/ext-bisect.md

```diff
@@ -0,0 +1,40 @@
+
+Finding a bad extension is easy: Open the extension view, [disable an extension]( https://code.visualstudio.com/docs/editor/extension-gallery#_disable-an-extension), reload, and check for the issue. If the problem is gone that extension is bad and you are done. Otherwise, re-enable the extension and repeat the process with the next extension.
+
+![Disabling an extension step by step](./disable_manually.png)
+
+If you are lucky the first extension is bad, if you are unlucky it’s the last extension. Using computer science language this means with `N` extensions you have a worst-case of `O(N)`, and an average-case of `O(N/2)`. Because this algorithm is operated by a human even small values of `N` are laborious and mindless. This is where extension bisect comes into play. It is much better in the worst- and average-case because it disables extensions by halves. This is like the problem of finding out which commit in git has caused a regression. Git has the [`git bisect`](https://git-scm.com/docs/git-bisect)-command for that and that inspired us to do the same for extensions.
+
+Let’s use a sample:  I have 24 extensions installed and the 8th extension is bad. We know that the iterative approach requires 8 steps. What about bisect?
+
+![The extension bisect process](bisect.gif)
+
+1. Bisect divides 24 extensions into two halves of 12 and 12 extensions and it disables all 12 extensions of the second half.
+2. In this sample the 8th extension is bad, it is the first half and not disabled. Things are still bad. Because of that bisect divides the first 12 extensions into 6 enabled and 6 disabled extensions. All other extensions are also re-enabled.
+3. The 8th extension is now disabled. Things are good now. That means bisect proceeds on the second half (extensions 6-11) and divides them into 3 enabled and 3 disabled extensions.
+4. Now, the 8th extension is re-enabled, and things have turned bad. This means bisect proceeds on the first half. It divides them into 1 enabled and 2 disabled extensions.
+5. The 8th extension is now disabled, things are good again and bisect proceeds on the second half, dividing it into 1 enabled and 1 disabled extension.
+6. The 8th extension is the only disabled extension and things are good. This means we have found the bad extension and that we are done.
+
+We see that in each step bisect reduces the search space by half. That means it runs in logarithmic time, resulting in average- and worst-case performance of `O(log N)`. That’s pretty good because it scales well. With 24 extensions you have 4 to 5 steps to find a bad extension, with 38 extensions it only takes 1 more step. However, best-case is worse because with the iterative approach you can be lucky and find the bad in the first round.
+
+Keep in mind that bisect relies on you giving correct feedback. You can easily fool it, and yourself, by always answering “Good now” (blames the last extension) or “This is bad” (won’t find an extension).
```

Comment:
I can't picture in my head how this works. Is there a screenshot that we can provide here?